translate/compiler_internal_test.go

```diff
@@ -891,10 +891,86 @@ func TestCompileIntegerLiteral_negative(t *testing.T) {
 
 // TODO: implement test cases :-)
 func TestCompileStringLiteral(t *testing.T) {
+	tests := []expressionCompileTestCase{
+		{
+			expression: &ast.StringLiteral{
+				Value: "string",
+			},
+			expected: Bytecode{
+				RawByte: []byte{0x21, 0x73, 0x74, 0x72, 0x69, 0x6e, 0x67, 0x00, 0x00, // Push 0x737472696e6700
+					0x21, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x08, // Push 0x0000000000000008 (size)
+					0x21, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, // Push 0x0000000000000000 (offset)
+					0x23}, // Memory store
+				AsmCode: []string{"Push", "737472696e670000",
+					"Push", "0000000000000008",
+					"Push", "0000000000000000",
+					"Mstore"},
+			},
+		},
+		{
+			expression: &ast.StringLiteral{
+				Value: "12345678",
+			},
+			expected: Bytecode{
+				RawByte: []byte{0x21, 0x31, 0x32, 0x33, 0x34, 0x35, 0x36, 0x37, 0x38, // Push 0x3132333435363738
+					0x21, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x08, // Push 0x0000000000000008 (size)
+					0x21, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, // Push 0x0000000000000000 (offset)
+					0x23}, // Memory store
+				AsmCode: []string{"Push", "3132333435363738",
+					"Push", "0000000000000008",
+					"Push", "0000000000000000",
+					"Mstore"},
+			},
+		},
+		{
+			expression: &ast.StringLiteral{
+				Value: "ssssssss1111111",
+			},
+			expected: Bytecode{
+				RawByte: []byte{0x21, 0x73, 0x73, 0x73, 0x73, 0x73, 0x73, 0x73, 0x73, // Push 0x7373737373737373
+					0x21, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x08, // Push 0x0000000000000008 (size)
+					0x21, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, // Push 0x0000000000000000 (offset)
+					0x21, 0x31, 0x31, 0x31, 0x31, 0x31, 0x31, 0x31, 0x00, // Push 0x3131313131313100
+					0x21, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x08, // Push 0x0000000000000008 (size)
+					0x21, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x08, // Push 0x0000000000000000 (offset)},
+					0x23, // Memory store
+				},
+				AsmCode: []string{"Push", "7373737373737373",
+					"Push", "0000000000000008",
+					"Push", "0000000000000000",
+					"Push", "3131313131313100",
+					"Push", "0000000000000008",
+					"Push", "0000000000000008",
+					"Mstore"},
```

Comment:
In this case, it just saves `3131313131313100`, not `7373737373737373`. 

Before executing `Mstore`, the stack would be
```asm
0000000000000008
0000000000000008
3131313131313100
0000000000000000
0000000000000008
7373737373737373
```
And, executing `Mstore`, it pops 3 data like `0000000000000008`, `0000000000000008`, `3131313131313100` and saves `3131313131313100` to memory. Then vm executes other opcode. 
Stack would be 
```asm
0000000000000000
0000000000000008
7373737373737373
```

So, I think `Mstore` is needed more. But, i know that we need to talk about longer string :)